src/main/groovy/lang/ObjectRange.java

```diff
@@ -397,7 +392,10 @@ public boolean contains(Object value) {
 
     @Override
     public void step(int step, Closure closure) {
-        if (step == 0 && compareTo(from, to) == 0) {
+        if (compareEqual(step, 0)) {
+            if (compareNotEqual(from, to)) {
+                throw new GroovyRuntimeException("Infinite loop detected due to step size of 0");
```

Comment:
That might be the case, but that does  not mean the other cases are right, not does it mean it should be done like that here as well. I think this should be looked at on a case-by-case base, as well as conceptually.